sky/serve/controller.py

```diff
@@ -120,8 +120,12 @@ async def update_service(request: fastapi.Request):
                 logger.info(
                     f'Update to new version version {version}: {service}')
 
-                self._replica_manager.update_version(version,
-                                                     service,
+                # TODO(Xuanlin Jiang): This assertion is disabled because of
+                # possibility of race condition.
+                # assert version == self._replica_manager.get_latest_version(
+                #    self._service_name)
```

Comment:
What does this mean? Can we elaborate this or is it necessary to have this?

Reply:
This is about what your first comment pointed out. I didn't realize tha the `if` code was to handle concurrent updates. Now it looks appropriate to use a similar `if` statement here as in deleted code mentioned in your first comment.

Reply:
The commented assertion is incorrect now by the way. It should be `serve_state.get_latest_verion(...)` on the right hand side.